File: src/Workers/SecurityPenetrationTestWorker/SanitizedEnvironment.py
#! /usr/bin/env python

"""

SanitizedEnvironment.py: checking if we are dealing with a sanitized environment

As part of the RemoteWorkers program which creates a framework for remote
management of laptops, desktop and servers. 

Copyright (C) 2012 Remote Workers Project. All rights reserved.
Created by Michal Parusinski <mparusinski@googlemail.com> on 18/05/2012.

"""

import sys
import os

def main(arguments):
    environ = os.environ
    homeVar = environ['HOME']
    userVar = environ['USER']
    if not homeVar == '/Users/' + userVar:
        print "The environment variables were changed"
        return 0
    else :
        print "The environment variables were not changed"
        return 1

if __name__ == "__main__":
    sys.exit(main(sys.argv))

File: src/Workers/SecurityPenetrationTestWorker/SanitizedEnvironment.cc
/* 

SanitizedEnvironment.cc: Check if the environment is sanitized

As part of the Remote Workers Project which creates a framework for remote
management of laptops, desktop and servers. 

Copyright (C) 2012 Remote Workers Project. All rights reserved.
Created by Michal Parusinski <mparusinski@googlemail.com> on 14/07/2012.

*/

#include "SanitizedEnvironment.h"

#include <cstdlib>
#include <cstdio>
#include <QProcess>

void checkForModificationsOfEnv()
{
    const QProcessEnvironment currentEnv = QProcessEnvironment::systemEnvironment();
    const QString homeValue = currentEnv.value("HOME");
    const QString userValue = currentEnv.value("USER");
    if ( (homeValue != "/Users/" + userValue) || (homeValue != "/Users/" + userValue + "/") ) {
        printf("MAC OS X detected or system environment has been modified\n");
    } else if ( (homeValue != "/home/" + userValue) || (homeValue != "/home/" + userValue + "/") ) {
        printf("Unix/Linux detected or system environment has been modified\n");
    } else {
        printf("System environment has been modified\n");
    } // WINDOWS NOT TREATED YET
}